<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<title>Documentation of VAG v1.01</title>
</head>
<body>
<div id="wmd-preview" class="wmd-preview"><div class="md-section-divider"></div><div class="md-section-divider"></div><h1 data-anchor-id="rkzr" id="documentation-of-vag-v101">Documentation of VAG v1.01</h1><p data-anchor-id="wrk0"></p><hr><div class="md-section-divider"></div><h3 data-anchor-id="7x9d" id="visualization-and-review-of-graphic-pan-genome-reads-alignment-based-on-vag-not-only-visualization-of-read-alignment">Visualization and review of graphic pan-genome reads alignment based on VAG (not only visualization of read alignment)</h3><p data-anchor-id="mgz0">The purpose of this document is to provide sufficient operational instructions for researchers to further integrate VAG services into scientific research and production.(<a href="https://github.com/lipingfangs/VAG" target="_blank">https://github.com/lipingfangs/VAG</a>)</p><p data-anchor-id="nh16">The graphical pan-genome supported by VAG can be constructed by minigraph, cactus and other tools. The GFA files generated by the construction tools (specific reference genome main line is required; In some cases, it needs to be sorted to S line and interleft with the corresponding P line). It needs to be further converted to the Fasta format file through gfatools.</p><div class="md-section-divider"></div><h3 data-anchor-id="u1xe" id="executable-version-of-window-systems">Executable version of Window systems</h3><p data-anchor-id="quve">With the further maturity of VAG, we have launched the window version of VAG, and the interface will be further optimized. At present, we can basically perform the work from Graph file extraction, bam file extraction to visual display (the current version only supports read alignment result display (read) and graph genome (Graph) itself). </p><p data-anchor-id="9hf5">The Window version has integrated all dependent packages, click-to-use, by generating local image files and calling the browser to display interactive images. However, the generation of &lt;.info&gt; file and the conversion of pan-genome map format still need to be generated by getinf.py script and the script processed by gfatools.</p><p data-anchor-id="kp3i">For the visualization process, the user can freely select the parameters on the panel after clicking the button named "Run the visualization" selected by graphsamtools. The generated interval extraction fold (index) can be visualized. The current file generates a PNG image by default and calls the user's local default browser to open the interactive format visualization image. <br>
<img src="src/20230228100438.jpg" alt="Example in window" title=""></p><p data-anchor-id="5jmu">For the sequence extraction process (graphsamtools), it is still recommended to carry out on Linux platform because the file size and computing resources consumed by the process are difficult for general Windows platform machines.</p><div class="md-section-divider"></div><h3 data-anchor-id="czcl" id="command-line-version-the-core-of-vag">Command line version (The core of VAG)</h3><p data-anchor-id="qkv4">The following "&lt;&gt;" shows the variable file naming content</p><div class="md-section-divider"></div><pre class="prettyprint linenums prettyprinted" data-anchor-id="6jsh" style=""><ol class="linenums"><li class="L0"><code><span class="pln">gfatools gfa2fa </span><span class="pun">-</span><span class="pln">s </span><span class="pun">&lt;</span><span class="typ">Graph</span><span class="pln"> name</span><span class="pun">&gt;.</span><span class="pln">gfa </span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;</span><span class="typ">Graph</span><span class="pln"> name</span><span class="pun">&gt;.</span><span class="pln">fa</span></code></li></ol></pre><p data-anchor-id="pgbg">The script  getinf.py  integrated in VAG can be used to convert the Fasta file of Graphs to the info file required by the VAG.</p><div class="md-section-divider"></div><pre class="prettyprint linenums prettyprinted" data-anchor-id="ex9h" style=""><ol class="linenums"><li class="L0"><code><span class="pln">python script</span><span class="pun">/</span><span class="pln">getinf</span><span class="pun">.</span><span class="pln">py </span><span class="pun">&lt;</span><span class="typ">Graph</span><span class="pln"> name</span><span class="pun">&gt;.</span><span class="pln">fa </span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;</span><span class="pln">info</span><span class="pun">.</span><span class="pln">document</span><span class="pun">&gt;</span></code></li></ol></pre><p data-anchor-id="fxuc">The further working process of VAG can be divided into two parts: <br>
● The graphsamtools script was used to extract the target interval of interest based on the selected reference genome interval.  <br>
● Using runVAG.py for visual production. This procedure outputs interactive format out.html and out.png files with default parameters, <br>
● Based on the data types of visulization this part can be divided into:</p><p data-anchor-id="vyst"><strong>1.Display of Graphic pan-genome:</strong></p><div class="md-section-divider"></div><pre class="prettyprint linenums prettyprinted" data-anchor-id="ln0t" style=""><ol class="linenums"><li class="L0"><code><span class="pln">graphsamtools </span><span class="pun">&lt;</span><span class="pln">info</span><span class="pun">.</span><span class="pln">document</span><span class="pun">&gt;</span><span class="pln">  </span><span class="pun">&lt;</span><span class="pln">chromosome number</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;</span><span class="pln">extraction start position</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;</span><span class="pln">extraction termination position</span><span class="pun">&gt;</span><span class="pln"> onlytrack </span><span class="pun">&lt;</span><span class="pln">output folder</span><span class="pun">&gt;</span></code></li><li class="L1"><code><span class="pln">python  runVAG</span><span class="pun">.</span><span class="pln">py </span><span class="pun">--</span><span class="pln">inindex  </span><span class="pun">&lt;</span><span class="pln">output folder of graphsamtools</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">--</span><span class="pln">drawtype onlytrack</span></code></li></ol></pre><p data-anchor-id="3q0e">An Example as follow:</p><p data-anchor-id="zvju"><img src="src/1677123117454.jpg" alt="1677123117454.jpg-41.7kB" title=""> <br>
You can also filter the branch display by the length of the branchs (--fl option）</p><div class="md-section-divider"></div><pre class="prettyprint linenums prettyprinted" data-anchor-id="cvci" style=""><ol class="linenums"><li class="L0"><code><span class="pln">python  runVAG</span><span class="pun">.</span><span class="pln">py </span><span class="pun">--</span><span class="pln">inindex  </span><span class="pun">&lt;</span><span class="pln">output folder of graphsamtools</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">--</span><span class="pln">drawtype onlytrack </span><span class="pun">--</span><span class="pln">fl </span><span class="pun">&lt;</span><span class="pln">filter length</span><span class="pun">&gt;</span></code></li></ol></pre><p data-anchor-id="jyyo">For example, the following is the effect of not showing branches below 10,000bp（branch）： <br>
<img src="src/1677123841485.png" alt="1677123841485.png-27kB" title=""></p><p data-anchor-id="1nh0"><strong>2. Display of alignment/mapping based on Graph pan-genome:</strong></p><p data-anchor-id="6puh">The Read to graph mapping phase can be handled with vg giraffe/map or other similar software. Where the generated .bam file is the input file for the VAG.</p><div class="md-section-divider"></div><pre class="prettyprint linenums prettyprinted" data-anchor-id="kv94" style=""><ol class="linenums"><li class="L0"><code><span class="pln">graphsamtools  </span><span class="pun">&lt;</span><span class="pln">info</span><span class="pun">.</span><span class="pln">document</span><span class="pun">&gt;</span><span class="pln">  </span><span class="pun">&lt;</span><span class="pln">chromosome number</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;</span><span class="pln">extraction start position</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;</span><span class="pln">extraction termination position</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;.</span><span class="pln">bam file</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;</span><span class="pln">output folder</span><span class="pun">&gt;</span></code></li><li class="L1"><code><span class="pln">python  runVAG</span><span class="pun">.</span><span class="pln">py </span><span class="pun">--</span><span class="pln">inindex   </span><span class="pun">&lt;</span><span class="pln">output folder of graphsamtools</span><span class="pun">&gt;</span></code></li></ol></pre><p data-anchor-id="5frb">The basic example as follows: By default, only the Insertion/Deletion will be displayed, and mapping information will only be displayed to a depth of 20x, without SNP.</p><p data-anchor-id="e4xv"><img src="src/1677392879797.png" alt="1677392879797.png-60.8kB" title=""></p><p data-anchor-id="7ejb">The users can choose to display the pair-end information in short sequence alignment. Based on this information, VAG can be used to estimate the reliable alignment path.</p><div class="md-section-divider"></div><pre class="prettyprint linenums prettyprinted" data-anchor-id="fjt7" style=""><ol class="linenums"><li class="L0"><code><span class="pln">python  runVAG</span><span class="pun">.</span><span class="pln">py </span><span class="pun">--</span><span class="pln">inindex   </span><span class="pun">&lt;</span><span class="pln">output folder of graphsamtools</span><span class="pun">&gt;</span><span class="pln">  </span><span class="pun">--</span><span class="pln">pairend </span><span class="lit">1</span><span class="pln">  </span><span class="pun">--</span><span class="pln">anntracks </span><span class="lit">1</span></code></li></ol></pre><p data-anchor-id="fnn9"><img src="src/1677124317886.png" alt="1677124317886.png-115.2kB" title=""> <br>
The Users can choose to display the alignment information as coverage information (calculated based on <br>
mosdepth).</p><div class="md-section-divider"></div><pre class="prettyprint linenums prettyprinted" data-anchor-id="4vlo" style=""><ol class="linenums"><li class="L0"><code><span class="pln">graphsamtools  </span><span class="pun">&lt;</span><span class="pln">info</span><span class="pun">.</span><span class="pln">document</span><span class="pun">&gt;</span><span class="pln">  </span><span class="pun">&lt;</span><span class="pln">chromosome number</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;</span><span class="pln">extraction start position</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;</span><span class="pln">extraction termination position</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;.</span><span class="pln">bam file</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;</span><span class="pln">output folder</span><span class="pun">&gt;</span></code></li><li class="L1"><code><span class="pln">python runVAG</span><span class="pun">.</span><span class="pln">py  </span><span class="pun">--</span><span class="pln">inindex  </span><span class="pun">&lt;</span><span class="pln">output folder of graphsamtools</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">--</span><span class="pln">drawtype coverage</span></code></li></ol></pre><p data-anchor-id="06v5">The basic example as follows (bar chart; Line chart is optional) : <br>
<img src="src/1677389842298.png" alt="1677389842298.png-22.7kB" title=""> <br>
The users can also select multiple samples in the same scale for coverage information display (more than three samples only support line chart display).</p><div class="md-section-divider"></div><pre class="prettyprint linenums prettyprinted" data-anchor-id="perl" style=""><ol class="linenums"><li class="L0"><code><span class="pln">graphsamtools </span><span class="pun">&lt;</span><span class="pln">info</span><span class="pun">.</span><span class="pln">document</span><span class="pun">&gt;</span><span class="pln">  </span><span class="pun">&lt;</span><span class="pln">info</span><span class="pun">.</span><span class="pln">document</span><span class="pun">&gt;</span><span class="pln">  </span><span class="pun">&lt;</span><span class="pln">chromosome number</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;</span><span class="pln">extraction start position</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;</span><span class="pln">extraction termination position</span><span class="pun">&gt;</span><span class="pln"> mutiplesamples </span><span class="pun">&lt;输出文件夹&gt;</span><span class="pln"> </span><span class="pun">&lt;</span><span class="pln">bam</span><span class="pun">文件列表&gt;</span></code></li><li class="L1"><code><span class="pln">python VAG</span><span class="pun">/</span><span class="pln">runVAG</span><span class="pun">.</span><span class="pln">py </span><span class="pun">--</span><span class="pln">inindex   </span><span class="pun">&lt;</span><span class="pln">output folder of graphsamtools</span><span class="pun">&gt;</span><span class="pln">  </span><span class="pun">--</span><span class="pln">drawtype mutiplesamples</span></code></li></ol></pre><p data-anchor-id="uteg"><img src="src/1677391157318.jpg" alt="1677391157318.jpg-19.6kB" title=""></p><div class="md-section-divider"></div><pre class="prettyprint linenums prettyprinted" data-anchor-id="7x9c" style=""><ol class="linenums"><li class="L0"><code><span class="com">#bam File List File Sample (does not include this line):</span></code></li><li class="L1"><code><span class="pln">sample1</span><span class="pun">.</span><span class="pln">bam</span></code></li><li class="L2"><code><span class="pln">sample2</span><span class="pun">.</span><span class="pln">bam</span></code></li></ol></pre><p data-anchor-id="9jvi"><strong>3. The Coveages frequency (PAV) between different populations can also be demonstrated.</strong> <br>
The input file is the .frq file after the group PAV frequency is processed by the plink --maf (currently only two groups are supported; By default, the chi-square test is marked (Pvalue&lt; 0.01) area of difference)</p><div class="md-section-divider"></div><pre class="prettyprint linenums prettyprinted" data-anchor-id="4yd0" style=""><ol class="linenums"><li class="L0"><code><span class="pln">graphsamtools </span><span class="pun">&lt;</span><span class="pln">info</span><span class="pun">.</span><span class="pln">document</span><span class="pun">&gt;</span><span class="pln">   </span><span class="pun">&lt;</span><span class="pln">info</span><span class="pun">.</span><span class="pln">document</span><span class="pun">&gt;</span><span class="pln">  </span><span class="pun">&lt;</span><span class="pln">chromosome number</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;</span><span class="pln">extraction start position</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;</span><span class="pln">extraction termination position</span><span class="pun">&gt;</span><span class="pln"> population </span><span class="pun">&lt;群体</span><span class="lit">1.frq</span><span class="pun">&gt;,&lt;群体名称</span><span class="lit">1</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;群体</span><span class="lit">2.frq</span><span class="pun">&gt;,&lt;群体名称</span><span class="lit">2</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">&lt;输出文件夹&gt;</span></code></li><li class="L1"><code><span class="pln">python VAG</span><span class="pun">/</span><span class="pln">runVAG</span><span class="pun">.</span><span class="pln">py  </span><span class="pun">--</span><span class="pln">inindex </span><span class="pun">&lt;</span><span class="pln">output folder of graphsamtools</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">--</span><span class="pln">drawtype populationfreq</span></code></li></ol></pre><p data-anchor-id="dk9v">Bar chart:</p><p data-anchor-id="9sk8"><img src="src/1677392268789.png" alt="1677392268789.png-48.6kB" title=""> <br>
Line chart:</p><div class="md-section-divider"></div><pre class="prettyprint linenums prettyprinted" data-anchor-id="3xs5" style=""><ol class="linenums"><li class="L0"><code><span class="pln">python VAG</span><span class="pun">/</span><span class="pln">runVAG</span><span class="pun">.</span><span class="pln">py  </span><span class="pun">--</span><span class="pln">inindex </span><span class="pun">&lt;</span><span class="pln">output folder of graphsamtools</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">--</span><span class="pln">drawtype populationfreq </span><span class="pun">--</span><span class="pln">popline </span><span class="lit">1</span></code></li></ol></pre><p data-anchor-id="mxor"><img src="src/1677392247855.png" alt="1677392247855.png-42.9kB" title=""></p><div class="md-section-divider"></div><pre class="prettyprint linenums prettyprinted" data-anchor-id="yf9e" style=""><ol class="linenums"><li class="L0"><code><span class="com"># Sample group frq file (does not include this line) :</span></code></li><li class="L1"><code><span class="pln"> CHR                             SNP   A1   A2          MAF  NCHROBS</span></code></li><li class="L2"><code><span class="pln">   </span><span class="lit">1</span><span class="pln">                        </span><span class="typ">Chr1_950</span><span class="pln">    A    G       </span><span class="lit">0.3237</span><span class="pln">      </span><span class="lit">414</span></code></li><li class="L3"><code><span class="pln">   </span><span class="lit">1</span><span class="pln">                       </span><span class="typ">Chr1_1050</span><span class="pln">    A    G      </span><span class="lit">0.04348</span><span class="pln">      </span><span class="lit">414</span></code></li><li class="L4"><code><span class="pln">   </span><span class="lit">1</span><span class="pln">                       </span><span class="typ">Chr1_1150</span><span class="pln">    A    G            </span><span class="lit">0</span><span class="pln">      </span><span class="lit">414</span></code></li><li class="L5"><code><span class="pln">   </span><span class="lit">1</span><span class="pln">                       </span><span class="typ">Chr1_1250</span><span class="pln">    A    G            </span><span class="lit">0</span><span class="pln">      </span><span class="lit">414</span></code></li><li class="L6"><code><span class="pln">   </span><span class="lit">1</span><span class="pln">                       </span><span class="typ">Chr1_1350</span><span class="pln">    A    G            </span><span class="lit">0</span><span class="pln">      </span><span class="lit">414</span></code></li><li class="L7"><code><span class="pln">   </span><span class="lit">1</span><span class="pln">                       </span><span class="typ">Chr1_1450</span><span class="pln">    A    G            </span><span class="lit">0</span><span class="pln">      </span><span class="lit">414</span></code></li><li class="L8"><code><span class="pln">   </span><span class="lit">1</span><span class="pln">                       </span><span class="typ">Chr1_1550</span><span class="pln">    A    G            </span><span class="lit">0</span><span class="pln">      </span><span class="lit">414</span></code></li><li class="L9"><code><span class="pln">   </span><span class="lit">1</span><span class="pln">                       </span><span class="typ">Chr1_1650</span><span class="pln">    A    G            </span><span class="lit">0</span><span class="pln">      </span><span class="lit">414</span></code></li><li class="L0"><code><span class="pln">   </span><span class="lit">1</span><span class="pln">                       </span><span class="typ">Chr1_1750</span><span class="pln">    A    G            </span><span class="lit">0</span><span class="pln">      </span><span class="lit">414</span></code></li></ol></pre><p data-anchor-id="pllb"><strong>4.Annotation file</strong></p><p data-anchor-id="22jn">Gene /TE and other site annotation files can be added to the VAG for joint visualization:</p><div class="md-section-divider"></div><pre class="prettyprint linenums prettyprinted" data-anchor-id="xmuy" style=""><ol class="linenums"><li class="L0"><code><span class="pln">python VAG</span><span class="pun">/</span><span class="pln">runVAG</span><span class="pun">.</span><span class="pln">py  </span><span class="pun">--</span><span class="pln">inindex </span><span class="pun">&lt;</span><span class="pln">output folder of graphsamtools</span><span class="pun">&gt;</span><span class="pln"> </span><span class="pun">--</span><span class="pln">drawtype populationfreq </span><span class="pun">--</span><span class="pln">gaingene </span><span class="lit">1</span><span class="pln"> </span><span class="pun">--</span><span class="pln">gff </span><span class="pun">&lt;</span><span class="pln">gff</span><span class="pun">格式的注释文件&gt;</span></code></li></ol></pre><p data-anchor-id="9fjt"><img src="src/0fb01ce48c2031759d597fedc14e6fa.jpg" alt="0fb01ce48c2031759d597fedc14e6fa.jpg-50.7kB" title=""></p><div class="md-section-divider"></div><h3 data-anchor-id="wwz7" id="web-version">Web version</h3><p data-anchor-id="udtu">WThe Web version supports deployment based on the Apache framework and relies on the php environment. Users can use theoutput folder of graphsamtools; After compression, upload the compressed package for visualization.</p><p data-anchor-id="q29a"><img src="src/%E5%9B%BE%E7%89%87_20230227151206.png" alt="图片_20230227151206.png-65.4kB" title=""> <br>
If you have any questions or comments on the use of this website, feel free to contact the author (lpf_bio@foxmali.com/fangping.li@scau.edu.cn)，we will reply promptly!</p><p data-anchor-id="eow9"><strong>This work was done by the molecular breeding team at South China Agricultural University in association with the Rice Institute of Guangdong Academy of Agricultural Sciences.</strong> If your research uses our tools, please cite the paper: Visualization and review of reads alignment on the graphical pan-genome with VAG Fangping Li, Haifei Hu, Zitong Xiao, Jingming Wang, Jieying Liu, Deshu Zhao, Yu Fu, Yijun Wang, Xue Yuan, Suhong Bu, Xiaofan Zhou, Junliang Zhao, Shaokui Wang bioRxiv 2023.01.20.524849; doi: <a href="https://doi.org/10.1101/2023.01.20.524849" target="_blank">https://doi.org/10.1101/2023.01.20.524849<a class="easyScholarDOI" target="_blank" href="https://sci-hub.wf/10.1101/2023.01.20.524849"><svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="22px" height="22px" viewBox="0 0 32 32" enable-background="new 0 0 32 32" xml:space="preserve">  <image id="image0" width="32" height="32" x="0" y="0" href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAABGdBTUEAALGPC/xhBQAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAABmJLR0QA/wD/AP+gvaeTAAAHdklEQVRYw+1WW2xcVxVd59zn3JnxPDwzHo89fsexYzvO02nSPBQgRMGqVEgFtEiABCriK0L8RHwgJFQJfpCgX6SFQgskqAlRcKAlhJa2zosSJ07i2LXj58Rjz4zHd9535r4OH3YSkiYQARI/WdLRlY7u2Xvdu89aewNP8AT/Z5DHeakmXMs3d25usKRApGBwzrJuc5ZtW7D0Cge9RCtqMp+YXEzGZ8v/UwKR+gaxofdTT6VQ/4mK4NsQqvaFo0GXFPTIkEUOFd20UpmSFkuk04uLySlWTAw5iuPnZ6+9P2Waxn9HYNOuAy2qa+PzXFV0394trU3PbG/x9TT7eb9bgihQEAIwBhimDTVfNkdm0tpbF6dTf7o4OppPTJ11ZYdOT1wdjP9HBNbv/lxvQtn4ze2be/YeOrgx3NdZI4s8BRjAAICxleedIISAEMAwLPP61FL55RNX4mfOD58LGRPHbg8NvJ/JqPajCHAPbnQ//Wx3yr31Wy/079j30te213Y2Vou5oo5c0UDZsHBpLImiZoIQAqfMgxDAshksm4HjKI0EXeKeDfWetMY3XIrRtWs61rmaq7nY3OxM8d8SaO3aUq36dx36woGnP/Pdr/QF/W4HDwDzSyUIPEWVIuDCaBK1fgVhv4KF5RKyRQNT8Rxup4sIemRwlECRBdoR9Urnx3OhyTTX1tvT3bB3Q21mbOyj+YpuPJqAs/PZg90bt77wgxd3NgQ8Dt5mDIQQpLJlKBIPr0uGIvHIlXQ011YhldVwYTQFReZBQOB3S5BFDowBHqdEP4oVxOuxoi9dFqLPP7O789Nbwk41OTc3u5Ap3cnJ3/369Tsjmrtx/9f7u2sjASdv2ytVZmCwbQaeoyiVDUwv5hDyOhBL5mFZDC6Zh5qvwCFx4DmKO5eDEgKRpyCE8gWDBGMFx1NfPfiNSMRV7Gg9dvTnr52J37iPgOnr3LZuTUPnng1RN2P3rphtAy4HD5eDR7FsQhQ4ZIo6qhQB7fUetEaq7v1OSsAAUEqQUDUMTS+DEsBmhLcYdQUCgfbN+1/0Sbd/5w14ml4fuMaP8wDQtrbbkxfD27Z1RcPVbpm3VwlQQgAKREMuMAZ4XRz29tZiVQgAAIEjYFgp1R0kMxp+fHIUN2czIIRAESk2rfGDUso73b5gS9S3u784JIQa+sd4n9frbO/pW3e57O9oi/rdIABhwFKujLFYFoQSVLsleJ0iFImHwFNwlNz1AdNmqOgWskUdC6qG69Mq3rocx7UpFZbNIPAUX9zdhF1dNSvVSQ/zkjblCzqxZi1JKHwo4AnvaDFad9c7wl31Th4gYITA65IQ8Mh4d3gRl8eXkcqvuKzMc5BEHhwlsC0bmm4hV9ahFnRkCwZKFROEAG6HgLa6Kjy3swGf3dEIhyzATA5Bu/Qd09IyhmYiu2RIizwjIoMcMvfXXEPr4t9RMXvBhbaDVjWjM+pFR4MPamFFajfnMpiM5zG/VIRa1KHpFizbBqUE9QEnuqMiwn4HWiJudDX6sDbqhc9JYOdmUBk7gcrIK6aZmSmn8hgbTvr/eGqaDROPx6N8sv/z6yvBvsOH9sl7tyuDspm8DMJJoP51oNXrwXnawbnrQCQ/GOeAyQToFoFhMdgMIATgKSByNgQYoFYOdiEOM30D5sIgrIVBWLlZs1yx1UQeIxfjwWOvXao6OXRjKkMAoKOzy11pee7whk19X/rZt3cFFSPGm/PvwLp9Flb6OlBZBqgIIvlAZD+I5AcRqwDeAUJ4MGYBpgam58DKS7C1FFBOgVXygM3AALNgcAu38sF3/zxd98aRgYlBNZOz7+sFG/d9eafq2fa9Vw/3b93TWyfbAGDqsEtx2Ooo7PQV2Ms3YWcnwbQEoGfBTA2wjRXts9VoBADlQTgHiOSF5YiYKdasjmQip0/8ZeInR48P3Pxn87vrA7NX3r4o9gT/cOT3wzW9rdUtHqfIM8qDuptAq5qAhgOAbYKZRbBKBqioYBUVTM8BZgnMNgFCQDgJENygsh+m4McrZxbLA+cmh0OlK7868UDy+6xYKxXtgGJPj2ccIU5y1fd1hJ08RykYWxU9W00gg0heECUM6mkG9XWAVveAC/SCC6wH9XeB87aBuurw5oV04YdHhyaU4q03Lpx+dUDX9X/djDKp+WK1S5z5cA5BTlRqetuCTlHgKGMPHlslxD6+CADTtMzj701kX/rFBzfF3MTruZFTv0kmFh86oXysHecSU8tehU6cu1VxxjOWp7XO6/C5ZcpxlILhkaCEgIGZiXRBf/nk1cSPfv3eIEtd/eny8PE347dj+qPOcQ/bzCWmln1C8eq1mFY4O5zi1bxB3Q6ROh08EXgOhMAG7i2tbOhTcTV//K/jqe//8oMbb79z8ZSy/OGRyfO/PaeVio8cRu5TwcMgiBJq1u5sLbvatlcF6je1N0Ua2xuD1SGfwimyIFQqppHKlqxbsXRmfHZxPp2I35BL0+cLc38bUZcSJh4DjzUVA4AvVC+J3mjQEjx+RiWFEULAGCN2pcQZOdXMxZeWF2e0x433BE9wB/8Aa3ZiVq1gDAMAAAAldEVYdGRhdGU6Y3JlYXRlADIwMjItMTItMDFUMTQ6NDQ6MDIrMDE6MDBW0C0AAAAAJXRFWHRkYXRlOm1vZGlmeQAyMDIyLTEyLTAxVDE0OjQ0OjAyKzAxOjAwJ42VvAAAAABJRU5ErkJggg=="></image></svg></a></a>,</p><p data-anchor-id="zuvz">A Chinese version of doucmentation:<a href="https://lipingfangs.github.io/VAGreadme.html" target="_blank">https://lipingfangs.github.io/VAGreadme.html</a></p></div>
</body>
</html>
